pyproject.toml

```
@@ -102,3 +102,31 @@ ignore = [
 
 # Ignore too many leading '#' for block comment
 "*/tests/*" = ["E266"]
+
+
+[tool.pyright]
+pythonVersion = "3.10"
+# DevGoal: "strict"
+typeCheckingMode = "standard"
+include = [
+  "icepyx",
+]
+exclude = [
+  "**/__pycache__",
+  "icepyx/tests",
+]
+# DevGoal: Remove all ignores
+ignore = [
+  "icepyx/quest/*",
+  "icepyx/core/APIformatting.py",
+  "icepyx/core/auth.py",
+  "icepyx/core/exceptions.py",
+  "icepyx/core/icesat2data.py",
+  "icepyx/core/is2ref.py",
+  "icepyx/core/read.py",
+  "icepyx/core/spatial.py",
+  "icepyx/core/temporal.py",
+  "icepyx/core/validate_inputs.py",
```

```suggestion
  "icepyx/core/is2ref.py",
  "icepyx/core/read.py",
  "icepyx/core/spatial.py",
```

FWIW, it looks like we could remove these few already!

---

Sorry that rendered so obnoxiously...

---

27f4641